core/src/test/java/io/lionweb/lioncore/java/serialization/NodePopulatorTest.java

```diff
@@ -0,0 +1,181 @@
+package io.lionweb.lioncore.java.serialization;
+
+import static org.junit.Assert.assertEquals;
+
+import io.lionweb.lioncore.java.LionWebVersion;
+import io.lionweb.lioncore.java.language.LionCoreBuiltins;
+import io.lionweb.lioncore.java.model.ClassifierInstanceUtils;
+import io.lionweb.lioncore.java.model.impl.DynamicNode;
+import io.lionweb.lioncore.java.self.LionCore;
+import io.lionweb.lioncore.java.serialization.data.SerializedChunk;
+import io.lionweb.lioncore.java.serialization.data.SerializedClassifierInstance;
+import java.util.Collections;
+import org.junit.Test;
+
+public class NodePopulatorTest {
+
+  @Test
+  public void populateReferenceToBuiltinsValueWithCorrectID() {
+    AbstractSerialization serialization =
+        SerializationProvider.getStandardJsonSerialization(LionWebVersion.v2024_1);
+    DeserializationStatus deserializationStatus =
+        new DeserializationStatus(Collections.emptyList(), serialization.getInstanceResolver());
+    NodePopulator nodePopulator =
+        new NodePopulator(
+            serialization, serialization.getInstanceResolver(), deserializationStatus);
+
+    SerializedChunk chunk =
+        new LowLevelJsonSerialization()
+            .deserializeSerializationBlock(
+                "{\n"
+                    + "  \"serializationFormatVersion\": \"2024.1\",\n"
+                    + "  \"languages\": [],\n"
+                    + "  \"nodes\": [\n"
+                    + "    {\n"
+                    + "      \"id\": \"my-node\",\n"
+                    + "      \"classifier\": {\n"
+                    + "        \"language\": \"LionCore-M3\",\n"
+                    + "        \"version\": \"2024.1\",\n"
+                    + "        \"key\": \"Property\"\n"
+                    + "      },\n"
+                    + "      \"properties\": [],\n"
+                    + "      \"containments\": [],\n"
+                    + "      \"references\": [\n"
+                    + "        {\n"
+                    + "          \"reference\": {\n"
+                    + "            \"language\": \"LionCore-M3\",\n"
+                    + "            \"version\": \"2024.1\",\n"
+                    + "            \"key\": \"Property-type\"\n"
+                    + "          },\n"
+                    + "          \"targets\": [\n"
+                    + "            {\n"
+                    + "              \"resolveInfo\": \"Boolean\",\n"
+                    + "              \"reference\": \"LionCore-builtins-Boolean\"\n"
```

Comment:
This is the id of LW-2023.1 Boolean, isn't it? Then, according to https://github.com/LionWeb-io/specification/issues/279 _Decision: Variant cc_, this should fail.

In any case, another test case where the node id and the resolveInfo contradict each other might be useful.

Reply:
Right, I corrected this test to use the ID for LW2024.1 Boolean. I then modified another test to use the ID for LW2023.1 Boolean and I verified that cause an exception